src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.js

```diff
@@ -79,7 +79,16 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
       loanToValue: 55,
 
       async getAprQuote(tokenId?: string): Promise<number> {
-        return 0.0381
+        if (tokenId == null) throw new Error('Getting wrapped native token not supported yet. ' + 'Explicitly pass in tokenId for the wrapped token.')
+
+        const edgeToken = wallet.currencyConfig.allTokens[tokenId]
+        const tokenAddress: string | void = edgeToken?.networkLocation?.contractAddress
```

Comment:
Should the cleaner be used here as well? There seems to be a whole sequence of things that can go wrong here, from `tokenId == null`, then `token == null` (not currently handled), and finally a failure to clean the address.

I wonder if a utility function would help here. Then the utility can just return `string | void`, and we can have one throw like "Token ${tokenId} does not have a contract address for AAVE" if it returns `void`.

Reply:
I am now leveraging the new `asTokenContractAddress` cleaner from the upstream branch. Addresses which are not included in AAVE's smart contract state will simply return `0` as a response, so we're safe there.